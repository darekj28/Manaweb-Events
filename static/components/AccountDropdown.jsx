var React = require('react');
var Link = require('react-router').Link;

export default class AccountDropdown extends React.Component {
	render() {
		return (
			<li className="dropdown">
              <a href="#" className="SearchNavBarGlyphicon dropdown-toggle" data-toggle="dropdown">
                <span className="glyphicon glyphicon-user"></span>
              </a>
              <ul className="dropdown-menu">
              {this.props.currentUser['first_name'] && <li><a href="#">{this.props.name}</a></li>}
              <li className="divider"></li>
              {this.props.currentUser.isAdmin && 
                <li> <a href = "/adminTools"> Admin Tools </a> </li>
              }

                <li><Link to="/settings">Settings</Link></li>
				<li><a href="/logout">Logout</a></li>
              </ul>
            </li>
		);
	}
}