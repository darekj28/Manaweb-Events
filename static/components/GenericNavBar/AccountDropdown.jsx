var React = require('react');
var Link = require('react-router').Link;
var AppActions = require('../../actions/AppActions.jsx');

export default class AccountDropdown extends React.Component {
  removeCurrentUser() {
      AppActions.removeCurrentUser();
  }
	render() {
		return (
			     <li className="dropdown">
              <a href="#" className="SearchNavBarGlyphicon dropdown-toggle" data-toggle="dropdown">
                <span className="glyphicon glyphicon-user"></span>
              </a>
              <ul className="dropdown-menu">
                {this.props.currentUser['userID'] && 
                  <li className="unclickable-dropdown">
                  <a className="unclickable-name">{this.props.name}</a></li>}
                <li className="divider"></li>
                {this.props.currentUser.isAdmin && 
                  <li> <a href = "/admin"> Admin Tools </a> </li>}
                <li><Link to="/settings">Account Settings</Link></li>
				        <li><Link to="/" onClick={this.removeCurrentUser.bind(this)}>Sign Out</Link></li>
              </ul>
          </li>
		);
	}
}