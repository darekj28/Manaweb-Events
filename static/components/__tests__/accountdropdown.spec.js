import AccountDropdown from '../AccountDropdown.jsx';

describe("AccountDropdown", () => {
	it("renders correctly", () => {
		const name = "Jay Kim";
		const ad = renderer.create(
			<AccountDropdown name={name}/>
			);
		expect(ad).toMatchSnapshot();
	});
});