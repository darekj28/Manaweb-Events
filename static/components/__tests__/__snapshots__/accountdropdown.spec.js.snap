exports[`AccountDropdown renders correctly 1`] = `
<li
  className="dropdown">
  <a
    className="dropdown-toggle"
    data-toggle="dropdown"
    href="#">
    <span
      className="glyphicon glyphicon-user" />
  </a>
  <ul
    className="dropdown-menu">
    <li>
      <a
        href="#">
        Jay Kim
      </a>
    </li>
    <li
      className="divider" />
    <li>
      <a
        href="/settings">
        Settings
      </a>
    </li>
    <li>
      <a
        href="#">
        Your Posts
      </a>
    </li>
    <li>
      <a
        href="/logout">
        Logout
      </a>
    </li>
  </ul>
</li>
`;
